src/public/pages/main.js

```diff
@@ -26,8 +26,12 @@ const Contacts = ({contacts}) =>
     Number of contacts: {U.length(contacts)}
   </div>
 
-export default U.withContext((_, {state}) =>
+export default U.withContext((props, {state}) =>
   <div>
     <h1>Main page</h1>
     <Contacts contacts={U.view(["contacts", L.define([])], state)}/>
-  </div>)
+    { typeof L.get( [ 'params', 'id' ], props ) !== 'undefined'
+      && <div>With ID: { L.get( [ 'params', 'id' ], props ) }</div>
+    }
```

Comment:
I agree with all of your suggestions. You touched on a couple of things I was uncertain about, so the clarification is helpful. I'll make, test and commit the changes shortly.